src/graphics/program-lib/programs/standard.js

```diff
@@ -235,12 +235,15 @@ var standard = {
     },
 
     _setMapTransform: function (codes, name, id, uv) {
-        codes[0] += "uniform vec4 texture_" + name + "MapTransform;\n";
+        const varName = `texture_${name}MapTransform`;
+        const checkId = id + uv * 100;
 
-        var checkId = id + uv * 100;
+        // upload a 3x2 matrix and manually perform the multiplication
+        codes[0] += `uniform vec3 ${varName}0;\n`;
+        codes[0] += `uniform vec3 ${varName}1;\n`;
         if (!codes[3][checkId]) {
-            codes[1] += "varying vec2 vUV" + uv + "_" + id + ";\n";
-            codes[2] += "   vUV" + uv + "_" + id + " = uv" + uv + " * texture_" + name + "MapTransform.xy + texture_" + name + "MapTransform.zw;\n";
+            codes[1] += `varying vec2 vUV${uv}_${id};\n`;
+            codes[2] += `   vUV${uv}_${id} = vec2(dot(uv${uv}, ${varName}0.xy) + ${varName}0.z, dot(uv${uv}, ${varName}1.xy) + ${varName}1.z).xy;\n`;
```

Comment:
it would probably save an instruction to use xyz in dot 3 and avoid the addition - like you had before with the matrix.
so instead of 
`dot(uv, var.xy) + var.z` you should do
`dot(vec3(uv, 1.0), var);`

Reply:
actually, not sure it will save anything at all. As vec3(uv, 1.0) will likely generate two move instructions, being the same as your code at the end.

Reply:
I don't think there is an instruction difference, but the resulting code is slightly more readable so I've made that change, thanks.